Set JWT token to localStorage from cookies (#47432)
mistercrunch/Airflow: docs/apache-airflow/core-concepts/auth-manager/index.rst

* You'd like to use an auth manager that leverages an identity provider from your preferred cloud provider.
* You have a private user management tool that is only available to you or your organization.

Authentication related BaseAuthManager methods
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

* ``get_user``: Return the signed-in user.
* ``get_url_login``: Return the URL the user is redirected to for signing in.

JWT token management by auth managers
-------------------------------------
The auth manager is responsible of creating the JWT token and pass it to Airflow UI. The protocol to exchange the JWT
token between the auth manager and Airflow UI is using cookies. The auth manager needs to save the JWT token in a
cookie named ``_token`` before redirecting to the Airflow UI. The Airflow UI will then read the cookie, save it and
delete the cookie.

.. code-block:: python

    from airflow.api_fastapi.auth.managers.base_auth_manager import COOKIE_NAME_JWT_TOKEN

    response = RedirectResponse(url="/")

>     response.set_cookie(COOKIE_NAME_JWT_TOKEN, "_token", secure=True)

Reviewer: ```suggestion
    response.set_cookie(COOKIE_NAME_JWT_TOKEN, token, secure=True)
```

This is confusing as `_token` is the name of the cookie, it looks like we are missing the token value in the cookie

Author: cool :)

* You'd like to use an auth manager that leverages an identity provider from your preferred cloud provider.
* You have a private user management tool that is only available to you or your organization.

Authentication related BaseAuthManager methods
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

* ``get_user``: Return the signed-in user.
* ``get_url_login``: Return the URL the user is redirected to for signing in.

> Cookie management related AuthManagers

Reviewer 2: ```suggestion
JWT token management by auth managers
```

> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Reviewer 2: Please use `------` to make it a sub section

Author: updated.

> When you write your own auth manager, you have to set the JWT token to a cookie key ``_token``.
> before redirecting to the Airflow UI. This is the key used by the Airflow UI to extract the JWT token from
> the cookie. Once token accessed by the UI, key will be removed from the cookie.
> refer the fab auth manager for more details. :class:`~airflow.providers.fab.www.views.FabIndexView`
> an example of setting a cookie in a response object is shown below:
> 
> .. code-block:: python
> 
>   from airflow.api_fastapi.auth.managers.base_auth_manager import COOKIE_NAME_JWT_TOKEN
> 
>   response = RedirectResponse(url="/")
>   response.set_cookie(COOKIE_NAME_JWT_TOKEN, "token", secure=True, samesite="Lax")
>   return response
> 
> 
> .. note::
>     Do not set the cookie parameter ``httponly`` to ``True``. Airflow UI needs to access the JWT token from the cookie.
> 

Reviewer 2: ```suggestion
The auth manager is responsible of creating the JWT token and pass it to Airflow UI. The protocol to exchange the JWT token between the auth manager and Airflow UI is using cookies. The auth manager needs to save the JWT token in a cookie named ``_token`` before redirecting to the Airflow UI. The Airflow UI will then read the cookie, save it and delete the cookie.

.. code-block:: python
  from airflow.api_fastapi.auth.managers.base_auth_manager import COOKIE_NAME_JWT_TOKEN
  
  response = RedirectResponse(url="/")
  response.set_cookie(COOKIE_NAME_JWT_TOKEN, "_token", secure=True, samesite="Lax")
  return response
  
.. note::
    Do not set the cookie parameter ``httponly`` to ``True``. Airflow UI needs to access the JWT token from the cookie.
```

* You'd like to use an auth manager that leverages an identity provider from your preferred cloud provider.
* You have a private user management tool that is only available to you or your organization.

> .. note::

Reviewer 2: Thanks for updating the doc, I think this is indeed very important. Could you please create a sub section of "Authentication related BaseAuthManager methods" and put it in there? It would make more sense

Author: Sure will add.

Author: Updated document now.
